cmake/LYWrappers.cmake

```diff
@@ -477,11 +477,23 @@ function(ly_parse_third_party_dependencies ly_THIRD_PARTY_LIBRARIES)
         if(${dependency_namespace} STREQUAL "3rdParty")
             if (NOT TARGET ${dependency})
                 list(GET dependency_list 1 dependency_package)
+                list(LENGTH dependency_list dependency_list_length)
                 ly_download_associated_package(${dependency_package})
-                find_package(${dependency_package} REQUIRED MODULE)
+                if (dependency_list_length GREATER 2)
+                    # There's an optional interface specified
+                    list(GET dependency_list 2 component)
+                    list(APPEND packages_with_components ${dependency_package})
```

Comment:
So the 'X11Interface' in this example is only needed for display purposes?

Reply:
What do you mean by "display purposes"?

From what I can tell, I can't call it `3rdParty::X11::xcb`  because it would create a circular `FindX11.cmake` loop. We'd be overriding the `FindX11.cmake` file that CMake provides with our custom one. But part of the point here is that our custom one forwards up to the `FindX11.cmake` that CMake provides. ie:
```
# some call to ly_add_library does:
RUNTIME_DEPENDENCIES
    PRIVATE
        3rdParty::X11::xcb

# LYWrappers.cmake then strips off the 3rdParty:: and calls:
find_package(X11 COMPONENTS xcb)

# That searches through ${CMAKE_MODULE_PATH} and finds cmake/3rdParty/FindX11.cmake, which makes the same call:
find_package(X11 COMPONENTS xcb) # infinite loop
```

To get around this, we could either:
* Name our package something different. This is what I chose. I'm following the `FindOpenGLInterface.cmake` precedent.
* Have our `FindX11.cmake` remove `cmake/3rdParty` from `${CMAKE_MODULE_PATH}` before calling `find_package(X11 COMPONENTS xcb)`. That way it finds the next one in the path.

Reply:
got it

Reply:
I just changed this to take the 2nd option instead. I think it is clearer to call the package `X11` than `X11Interface`.